server/pulp/server/agent/direct/pulpagent.py

```diff
@@ -66,16 +68,15 @@ def cancel(context, task_id):
         :param task_id: The ID of a task associated with an agent request.
         :type task_id: str
         """
-        criteria = {'match': {'task_id': task_id}}
-
-        agent = Agent(
-            context.url,
-            context.route,
-            authenticator=context.authenticator,
-            wait=0)
-
-        admin = agent.Admin()
-        admin.cancel(criteria=criteria)
+        with context:
```

Comment:
The variable context could be more clearly named. I already know it's a context because of how it's being used so I really have no idea what this does.